src/common/core/metaquerycore.js

```diff
@@ -98,21 +98,10 @@ define([
                 delete rules.max;
                 delete rules.min;
 
-                //we need to clear nodes that are not on the meta sheet
-                // and we have to initialize the counters
-                keys = Object.keys(rules);
-                for (i = 0; i < keys.length; i += 1) {
-                    if (metaNodes[keys[i]]) {
-                        typeCounters[keys[i]] = 0;
-                    } else {
-                        delete rules[keys[i]];
-                    }
-                }
-
                 keys = Object.keys(rules);
                 for (i = 0; i < children.length; i += 1) {
                     for (j = 0; j < keys.length; j += 1) {
-                        if (innerCore.isTypeOf(children[i], metaNodes[keys[j]])) {
+                        if (innerCore.isTypeOfPath(children[i], keys[j])) {
```

Comment:
Can't this method be exposed on the core? Or alternatively `isTypeOf` could accept both a node and path as the second argument. What definitely should be done is to merge the implementation of these two.